src/modm/platform/dma/stm32/module.lb

```diff
@@ -16,15 +17,46 @@ def init(module):
     module.name = ":platform:dma"
     module.description = "Direct Memory Access (DMA)"
 
+
 def prepare(module, options):
     device = options[":target"]
-    if not device.has_driver("dma:stm32*"):
-        return False
 
-    # FIXME the driver is for L4 only
-    if device.identifier["family"] not in ["l4"]:
+    if (not device.has_driver("dma")) or device.get_driver("dma")["type"] not in \
+            [
+            "stm32-channel-request",
+            "stm32-channel",
+    ]:
         return False
-    if device.identifier["name"] not in ["12", "22", "31", "32", "33", "42", "43", "51", "52", "62", "75", "76", "86", "96", "A6"]:
+
+    did = device.identifier
+
+    # Enable DMA for all but some devices...
+    if not (
+        (did["family"] in ["f0"]
+         # STM32F09x has shared interrupts between DMA1 and DMA2, not supported...
+         and did["name"] not in ["30", "71", "72", "78", "91", "98"])
+        or (did["family"] in ["f1"] and
+            (did["name"] in ["02", "05", "07"]
+             or
+             (did["name"] in ["00", "01", "03"]
+              and did["pin"] in ["r", "v", "z"]
+              and did["size"] in ["c", "d", "e", "f", "g"]
+             )
+            )
+           )
+        or (did["family"] in ["f3"] and
+            did["name"] not in ["73", "78"]
+            and
+            (did["name"] in ["02"]
+             and did["pin"] not in ["c"]
+             and did["size"] not in ["b", "c"]
+             )
+            )
```

Comment:
The condition for F3 is not correct. It selects only a subset of F302 and excludes all others.

Did you mean
```
did["family"] in ["f3"] and
  (did["name"] not in ["02", "73", "78"]
    or
    (did["name"] in ["02"]
      and did["pin"] not in ["c"]
      and did["size"] not in ["b", "c"]
    )
   )
```
?

Reply:
Yeah, I'm going to split the whole if clause into multiple for better readability.

Reply:
See https://github.com/modm-io/modm/pull/608/files#diff-30b70740138f96c7450b9c83c30af2aaff822ccd83dd86c6e7fcd901a1650b93R35-R65